packages/fuel-indexer/src/executor.rs

```diff
@@ -64,12 +65,19 @@ impl ExecutorSource {
     }
 }
 
+/// Run the executor task until the kill switch is flipped, or until some other
+/// stop criteria is met.
+///
+/// In general the logic in this function isn't very idiomatic, but that's because
+/// types in `fuel_core_client` don't compile to WASM.
```

Comment:
We can probably change these two lines to be a regular comment instead of a doc comment as a user won't see the function logic in the crate documentation without clicking `Source`.